python/sidecar/http.py

```diff
@@ -192,20 +193,13 @@ async def create_project(project_name: str, project_path: str = None):
 
 
 @project_api.get("/{project_id}")
-async def get_project(project_id: str):
+async def get_project(project_id: str) -> ProjectFileTreeResponse:
```

Comment:
This type can work for the /files endpoint, but don't work for the "project details" response.
Also, I think you should split getting the project files from the project information (id, name, ...).

I'm also I on dropping the project path completely from the output.
Note that in the future we might (will) have extra fields/props for a project like last updated date and alikes.